src/DataAPI.jl

```diff
@@ -522,4 +522,14 @@ Return the row number of `row` in the source table.
 """
 function rownumber end
 
+"""
+    groupby(obj, args...; kw...)
+
+Group `obj` into partitions specified by `args` and `kw`,
+which may vary according to the implementation for `obj`.
+
+The return value an iterable over the groups.
```

Comment:
Why do we want to add this restriction to the contract? What if some package wants to e.g. return something that is indexable but not iterable?

Reply:
I assumed iterable is usually a subset of indexible, and a common property of the return value might make the interface more unified. But I can just remove this line.